include/boost/gil/image_processing/filter.hpp

```diff
@@ -60,6 +69,71 @@ void blur(
     box_filter(src_view, dst_view, kernel_size, anchor, true, option);
 }
 
+
+namespace detail
+{
+template <typename SrcView, typename DstView>
+void impl_median(SrcView const& src_view, DstView const& dst_view, std::size_t kernel_size)
+{
+    std::size_t half_kernel_size = kernel_size / 2;
+
+    // deciding output channel type and creating functor
+    using src_channel_t = typename channel_type<SrcView>::type;
+
+    for (std::ptrdiff_t y = 0; y < src_view.height(); y++)
+    {
+        typename DstView::x_iterator dst_it = dst_view.row_begin(y);
+
+        for (std::ptrdiff_t x = 0; x < src_view.width(); x++)
+        {
+            auto sub_view = subimage_view(
+                src_view,
+                x - half_kernel_size, y - half_kernel_size,
+                kernel_size,
+                kernel_size
+            );
+            std::vector<src_channel_t> values(sub_view.begin(), sub_view.end());
```

Comment:
I think this `vector` is likely/potentially reallocated many times but unnecessarily since the size upper limit is known and constant.

Do you agree with slight optimisation like this?

```
diff --git a/include/boost/gil/image_processing/filter.hpp b/include/boost/gil/image_processing/filter.hpp
index 32f4e6c4a..1ba6b2f38 100644
--- a/include/boost/gil/image_processing/filter.hpp
+++ b/include/boost/gil/image_processing/filter.hpp
@@ -80,6 +80,8 @@ void impl_median(SrcView const& src_view, DstView const& dst_view, std::size_t k
     // deciding output channel type and creating functor
     using src_channel_t = typename channel_type<SrcView>::type;

+    std::vector<src_channel_t> values;
+    values.reserve(kernel_size * kernel_size);
     for (std::ptrdiff_t y = 0; y < src_view.height(); y++)
     {
         typename DstView::x_iterator dst_it = dst_view.row_begin(y);
@@ -92,7 +94,7 @@ void impl_median(SrcView const& src_view, DstView const& dst_view, std::size_t k
                 kernel_size,
                 kernel_size
             );
-            std::vector<src_channel_t> values(sub_view.begin(), sub_view.end());
+            values.assign(sub_view.begin(), sub_view.end());

             std::nth_element(values.begin(), values.begin() + (values.size() / 2), values.end());
             dst_it[x] = values[values.size() / 2];
```

The `values.assign` will compare current capacity with new size required, and will not re-allocate.

Reply:
I was thinking something similar but creating a vector in the loop didn't feel right. Thank you for pointing this. I corrected it.